internal/proxy/ui_requests.go

```diff
@@ -116,17 +121,59 @@ func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request, r
 			if !ok {
 				continue
 			}
-			if strings.HasPrefix(cookie.Name, SessionCookie) { //We found a session cookie in cache
-				cacheKey = cookie.Value
+
+			if isSessionCookie(cookie) {
+				//We found a session cookie in cache
+				// If jenkins is not "running", return loading page and status 202
 				pci := cacheVal.(CacheItem)
-				r.Host = pci.Route //Configure proxy upstream
-				r.URL.Host = pci.Route
-				r.URL.Scheme = pci.Scheme
 				ns = pci.NS
-				needsAuth = false //user is probably logged in, do not redirect
-				noProxy = false
+				cacheKey = cookie.Value
+
+				requestLogEntry.WithField("ns", ns).
+					Infof("Cached Jenkins route %s in %s", pci.Route, cookie.Value)
+
+				state, err := p.idler.State(ns, pci.ClusterURL)
+				if err != nil {
+					p.HandleError(w, err, requestLogEntry)
+					return
+				}
+				if state != clients.Running {
+					// we find a session cookie but the pod isn't running
+					// so lets unidle and clear the cookie and show loading page
+					code, err := p.idler.UnIdle(ns, pci.ClusterURL)
+					if err != nil {
+						p.HandleError(w, err, requestLogEntry)
+						return
+					}
+
+					if code == http.StatusOK {
+						code = http.StatusAccepted
+					} else if code != http.StatusServiceUnavailable {
+						p.HandleError(w, fmt.Errorf("Failed to send unidle request using fabric8-jenkins-idler"), requestLogEntry)
+					}
+
+					w.WriteHeader(code)
+					clearCookiesMatching(w, r, isSessionCookie)
```

Comment:
I think JSESSIONID is the only cookie that is being created multiple times, right?
JenkinsIdled cannot be, because it will always have the same name.